src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,378 @@
+            'mailTheme' => $mailTheme,
+        ]);
+    }
+
+    /**
+     * Preview a mail layout from a defined theme
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param string $theme
+     * @param string $layout
+     * @param string $type
+     * @param string $locale
+     * @param string $module
+     *
+     * @return Response
+     *
+     * @throws FileNotFoundException
+     * @throws InvalidArgumentException
+     */
+    public function previewLayoutAction($theme, $layout, $type, $locale, $module = '')
+    {
+        $renderedLayout = $this->renderLayout($theme, $layout, $type, $locale, $module);
+
+        return new Response($renderedLayout);
+    }
+
+    /**
+     * Display the raw source of a theme layout (mainly useful for developers/integrators)
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param string $theme
+     * @param string $layout
+     * @param string $type
+     * @param string $locale
+     * @param string $module
+     *
+     * @return Response
+     *
+     * @throws FileNotFoundException
+     * @throws InvalidArgumentException
+     */
+    public function rawLayoutAction($theme, $layout, $type, $locale, $module = '')
+    {
+        $renderedLayout = $this->renderLayout($theme, $layout, $type, $locale, $module);
+
+        $response = new Response($renderedLayout, 200, [
+            'Content-Type' => 'text/plain',
+        ]);
+
+        return $response;
+    }
+
+    /**
+     * Dynamically display an email template, this is usually used by the MailGenerator but this action
+     * allows to display preview before generating the file (handy when you are developing an email theme)
+     *
+     * @param string $themeName
+     * @param string $layoutName
+     * @param string $type
+     * @param string $locale
+     * @param string $module
+     *
+     * @return string
+     *
+     * @throws FileNotFoundException
+     * @throws InvalidArgumentException
+     */
+    private function renderLayout($themeName, $layoutName, $type, $locale = '', $module = '')
+    {
+        /** @var ThemeCatalogInterface $themeCatalog */
+        $themeCatalog = $this->get('prestashop.core.mail_template.theme_catalog');
+        /** @var ThemeInterface $theme */
+        $theme = $themeCatalog->getByName($themeName);
+
+        /** @var LayoutInterface $layout */
+        $layout = null;
+        /* @var LayoutInterface $layoutInterface */
+        foreach ($theme->getLayouts() as $layoutInterface) {
+            if ($layoutInterface->getName() == $layoutName
+                && $layoutInterface->getModuleName() == $module
+            ) {
+                $layout = $layoutInterface;
+                break;
+            }
+        }
+
+        if (null === $layout) {
+            throw new FileNotFoundException(sprintf(
+                'Could not find layout %s%s in theme %s',
```

Comment:
Is it a wording? If yes, it needs to be localized. So I would write _Cannot find layout %s%s in theme %s_ in _Admin.Design.Notification_.

Reply:
Well this is the exception message when something goes wrong, it's not supposed to happen nor displayed so I don't think we need to translate these, but I'll use your text to harmonize the `Cannot` use